scripts/bq_load.sh

```diff
@@ -62,7 +62,20 @@ for bucket_prefix in `gsutil ls "$RESULT_BUCKET"`; do
     union="$union$subquery"
 done
 
-query="CREATE OR REPLACE TABLE \`$PROJECT_ID.$DEST_DATASET.$DEST_TABLE\` LIKE \`$PROJECT_ID.$LOAD_DATASET.$table_name\` PARTITION BY TIMESTAMP_TRUNC(CreatedTimestamp, DAY) OPTIONS(expiration_timestamp=NULL) AS $union;"
+# Query to populate the destination table.
+#
+# If the table does not exist it will be created. Keeping the table ensures
+# that breaking schema changes are not accidentally propogated to the
+# destination table.
+#
+# A transaction is used to keep the update atomic.
+query="
+CREATE TABLE IF NOT EXISTS \`$PROJECT_ID.$DEST_DATASET.$DEST_TABLE\` LIKE \`$PROJECT_ID.$LOAD_DATASET.$table_name\`
+    PARTITION BY TIMESTAMP_TRUNC(CreatedTimestamp, DAY) OPTIONS(expiration_timestamp=NULL);
+BEGIN TRANSACTION;
+TRUNCATE TABLE \`$PROJECT_ID.$DEST_DATASET.$DEST_TABLE\`;
```

Comment:
Is the point that the transaction (including truncation) will fail due to the insert if there are any breaking schema changes?

May be worth adding this explicitly on line 71

Reply:
Done!